src/store/modules/adminKPI.js

```diff
@@ -197,38 +197,85 @@ const state = {
-            kpi_name:payload.kpi_name,
-            kpi_json : [
+            kpi_name: payload.kpi_name,
+            kpi_json: [
                 {
-                    title : " ",
-                    desc  : " "
+                    title: " ",
+                    desc: " ",
+                    edit: "false",
+                    addKpi: true
                 }
             ],
-            kra_json : [
+            kra_json: [
                 {
-                    title : " ",
-                    desc  : " "
+                    title: " ",
+                    desc: " ",
+                    edit: "false",
+                    addEra: true
                 }
             ]
         }
-        await axios.post('http://5.9.144.226:8000/kpi',data,{
-            headers:{
-                "Content-Type" : 'application/json',
-                "Authorization" : `Bearer ${payload.token}`
+        await axios.post('http://5.9.144.226:8000/kpi', data, {
+            headers: {
+                "Content-Type": 'application/json',
+                "Authorization": `Bearer ${payload.token}`
             }
         }).then(response => {
-            dispatch('getKpiEra', {kpiEraId:response.data, token: payload.token})
+            dispatch('getKpiEra', { kpiEraId: response.data, token: payload.token })
         })
     },
-    async getKpiEra({commit},data){
-        await axios.get('http://5.9.144.226:8000/kpi',{
-            headers:{
-                "Authorization" : `Bearer ${data.token}`
+    async getKpiEra({ commit }, data) {
+        await axios.get('http://5.9.144.226:8000/kpi', {
+            headers: {
+                "Authorization": `Bearer ${data.token}`
             }
         }).then(response => {
-            commit('addNewTeam',response.data)
+            commit('addNewTeam', response.data)
+        })
+    },
+    async updateKpi({ commit, dispatch }, payload) {
+        let data = {
+            kpi_name: payload.data.kpi_name,
+            kpi_json: [
+                {
+                    title: payload.data.kpi_json[0].title,
+                    desc: payload.data.kpi_json[0].desc,
+                    edit: "false"
+                }
+            ],
+            kra_json: [
+                {
+                    title: "xxxx",
+                    desc: "xxxx",
+                    edit: "false"
+
+                }
+            ]
+        }
+        await axios.put(`http://5.9.144.226:8000/kpi/${payload.data._id}`, data, {
+            headers: {
+                "Content-Type": 'application/json',
+                "Authorization": `Bearer ${payload.Authorization}`
+            }
+        }).then(response => {
+            console.log(response);
+        }).catch(e => {
+            console.log(e);
+            dispatch('getKpiEra', { token: payload.Authorization })
+        })
+    },
+    async delKpi({ commit, dispatch }, data) {
+        await axios.delete(`http://5.9.144.226:8000/kpi/${data.id}`, {
```

Comment:
i had mentioned axios authorization will be global. it will not be there in all apis